kernel/Kconfig

```diff
@@ -180,6 +180,13 @@ config IDLE_STACK_SIZE
 	  logging (e.g. logging when the system goes to sleep), the idle thread
 	  may need more stack space than the default value.
 
+config ISR_SHARE_IRQ
+	bool "Allow IRQs to share multiple ISRs"
+	depends on ARCH_HAS_SW_ISR_TABLE_SHARED_IRQ
+	help
+	  Some peripherals share an IRQ and this option enables the system
+	  to call all ISRs attached to a pending IRQ.
```

Comment:
Why is this not placed in the `Interrupt Configuration` menu here:
https://github.com/zephyrproject-rtos/zephyr/blob/460ac477c1e9fcf36cee82c4d3c0571476a4bca2/arch/Kconfig#L380-L389

Secondly, should this setting be dependent on `GEN_ISR_TABLES` ?